### PR TITLE
Initial commit for gitops deploy orb

### PR DESCRIPTION
**Pull request by stephen-harris**

### CODE REVIEW

**Comment by stephen-harris** [comment]
@jnogol The current implementation assumes: `<manifest_directory>/values-<environment>-<region>.yaml`. I had initial thought of replacing region/environment parameters with just one to describe the path values file. However, this slightly more opinionated approach allows us to make it clear in the git commit which environment & region is being deployed to.

**Comment by stephen-harris** [comment]
One question is, do we need to support updating multiple paths? (e.g. multiple services/image versions)

**Review by gordonkeogh — APPROVED** [review]
LGTM

File: gitops/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+## Initial set-up
+
+Since this orb commits changes to the target Git repository (by default the repository checked out by CircleCI) it requires write-access to that repository. By default CircleCI has read-only access.
+
+### Create read-write deployment key
+
+#### Create a private/public key pair:
+
+Replace `<repo>` with the name of your repository.
```
Comment (jnogol) [inline gitops/README.md:17]: Small thingy: there's no `<repo>` in this section.

File: gitops/orb.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    description: "Install python dependencies"
+    steps:
+      - run:
+          name: "Install python dependencies"
+          command: |
+            pip3 install --upgrade pip
+            pip3 install requests
+
+  update_helm:
+    <<: *update-helm-params
```
Comment (jnogol) [inline gitops/orb.yml:68]: noob question, do you need to close the opening `<<` here?

File: gitops/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  cd:
+    jobs:
+      ...
+      gitops/update-helm:
+        name: gitops-deploy-sandbox-eu1
+        environment: sandbox
+        region: eu1
+        image_tag_yaml_path: .backend.application.image.tag
+        image_tag: $CIRCLE_SHA1
+        values_file: manifests/values-sandbox-eu1.yaml # This is teh default value for the given environment & region params
```
Comment (jnogol) [inline gitops/README.md:93]: Tiny typo: This is **teh** default value...

File: gitops/orb.yml (hunk 2 above)
Comment (stephen-harris) [reply gitops/orb.yml:69]: yes, its for injecting `update-helm-params`

**Review by RostamKal — APPROVED** [review]

**Pull request merged**